conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureHandler.java

```diff
@@ -32,44 +32,59 @@
             // no-cache and web-security handlers add listeners for the response to be committed,
             // they can be executed on the IO thread.
-            (endpoint, handler) -> Methods.GET.equals(endpoint.method())
+            endpoint -> handler -> Methods.GET.equals(endpoint.method())
                     // Only applies to GET methods
                     ? new NoCachingResponseHandler(handler) : handler,
-            (endpoint, handler) -> new WebSecurityHandler(handler),
-            // It is vitally important to never run blocking operations on the initial IO thread otherwise
-            // the server will not process new requests. all handlers executed after BlockingHandler
-            // use the larger task pool which is allowed to block. Any operation which sets thread
-            // state (e.g. SLF4J MDC or Tracer) must execute on the blocking thread otherwise state
-            // will not propagate to the wrapped service.
-            (endpoint, handler) -> new BlockingHandler(handler),
-            // Logging context and trace handler must execute prior to the exception
-            // to provide user and trace information on exceptions.
-            (endpoint, handler) -> new LoggingContextHandler(handler),
-            (endpoint, handler) -> new TracedOperationHandler(
-                    handler, endpoint.method() + " " + endpoint.template()),
-            (endpoint, handler) -> new ConjureExceptionHandler(handler)
+            endpoint -> handler -> new WebSecurityHandler(handler)
+    ).reverse();
+    // It is vitally important to never run blocking operations on the initial IO thread otherwise
+    // the server will not process new requests. all handlers executed after BlockingHandler
+    // use the larger task pool which is allowed to block. Any operation which sets thread
+    // state (e.g. SLF4J MDC or Tracer) must execute on the blocking thread otherwise state
+    // will not propagate to the wrapped service.
+    private static final ImmutableList<EndpointHandlerWrapper> WRAPPERS_AFTER_BLOCKING =
+            ImmutableList.<EndpointHandlerWrapper>of(
+            endpoint -> handler -> new BlockingHandler(handler),
+    // Logging context and trace handler must execute prior to the exception
+    // to provide user and trace information on exceptions.
+            endpoint -> handler -> new LoggingContextHandler(handler),
+            endpoint -> handler -> new TracedOperationHandler(
+            handler, endpoint.method() + " " + endpoint.template()),
+            endpoint -> handler -> new ConjureExceptionHandler(handler)
     ).reverse();
 
+    public ConjureHandler addHandlerWrapperBeforeBlocking(EndpointHandlerWrapper wrapper) {
+        wrappersJustBeforeBlocking = ImmutableList.<EndpointHandlerWrapper>builder()
+                .addAll(wrappersJustBeforeBlocking)
+                .add(wrapper)
+                .build();
+        return this;
+    }
```

Comment:
ConjureHandler is meant to be immutable, configuration should occur on the builder

Maybe call the builder method `initialNonBlockingEndpointHandlerWrappers(EndpointHandlerWrapper)` with strongly worded documentation warning against performing _any_ blocking operations in this wrapper. We should also call out the order we expect them to be invoked in.

Reply:
Added doc, but kept the name.